src/wp-admin/includes/taxonomy.php

```diff
@@ -188,7 +188,7 @@ function wp_insert_category( $catarr, $wp_error = false ) {
 function wp_update_category( $catarr ) {
 	$cat_id = (int) $catarr['cat_ID'];
 
-	if ( isset( $catarr['category_parent'] ) && ( $cat_id == $catarr['category_parent'] ) ) {
+	if ( isset( $catarr['category_parent'] ) && ( $cat_id === $catarr['category_parent'] ) ) {
```

Comment:
This function is an alias for `wp_insert_category()`, which accepts a `category_parent` of `int|string`:
```
category_parent int|string
Category parent ID. Default empty.
```
[Reference](https://developer.wordpress.org/reference/functions/wp_insert_category/).

To protect backward compatibility, `$catarr['category_parent']` should be cast to `int`:
```suggestion
	if ( isset( $catarr['category_parent'] ) && ( $cat_id === (int) $catarr['category_parent'] ) ) {
```